web/src/api/images/verify_app.ts

```diff
@@ -0,0 +1,278 @@
+    const sourcePrefix = `unverified/${app_id}/`;
+    const destinationPrefix = `verified/${app_id}/`;
+
+    // In case we have some stale images in the folder we iterate directly to delete
+    const listObjectsResponse = await s3Client.send(
+      new ListObjectsCommand({
+        Bucket: bucketName,
+        Prefix: destinationPrefix,
+      })
+    );
+
+    const verifiedImageKeysToDelete = listObjectsResponse.Contents?.map(
+      (object) => object.Key
+    );
+
+    if (verifiedImageKeysToDelete && verifiedImageKeysToDelete.length > 0) {
+      const expirePromises = verifiedImageKeysToDelete.map((key) =>
+        s3Client.send(
+          new PutObjectTaggingCommand({
+            Bucket: bucketName,
+            Key: key,
+            Tagging: {
+              TagSet: [
+                {
+                  Key: "expired",
+                  Value: "true",
+                },
+              ],
+            },
+          })
+        )
+      );
+      await Promise.all(expirePromises);
+    }
+
+    // Copy unverified images to verified images with random names
+    const copyPromises = [];
+
+    const currentLogoImgName = awaitingReviewAppMetadata.logo_img_url;
+    const logoFileType = getFileExtension(currentLogoImgName);
+    const newLogoImgName = randomUUID() + logoFileType;
+    copyPromises.push(
+      s3Client.send(
+        new CopyObjectCommand({
+          Bucket: bucketName,
+          CopySource: `${bucketName}/${sourcePrefix}${currentLogoImgName}`,
+          Key: `${destinationPrefix}${newLogoImgName}`,
+        })
+      )
+    );
+
+    const currentHeroImgName = awaitingReviewAppMetadata.hero_image_url;
+    const heroFileType = getFileExtension(currentHeroImgName);
+    const newHeroImgName = randomUUID() + heroFileType;
+    copyPromises.push(
+      s3Client.send(
+        new CopyObjectCommand({
+          Bucket: bucketName,
+          CopySource: `${bucketName}/${sourcePrefix}${currentHeroImgName}`,
+          Key: `${destinationPrefix}${newHeroImgName}`,
+        })
+      )
+    );
+
+    const showcaseImgUrls = awaitingReviewAppMetadata.showcase_img_urls;
+    const showcaseFileTypes = showcaseImgUrls.map((url: string) =>
+      getFileExtension(url)
+    );
+    const showcaseImgUUIDs = showcaseImgUrls.map(
+      (_: string, index: number) => randomUUID() + showcaseFileTypes[index]
+    );
+    const showcaseCopyPromises = showcaseImgUrls.map(
+      (key: string, index: number) => {
+        return s3Client.send(
+          new CopyObjectCommand({
+            Bucket: bucketName,
+            CopySource: `${bucketName}/${sourcePrefix}${key}`,
+            Key: `${destinationPrefix}${showcaseImgUUIDs[index]}`,
+          })
+        );
+      }
+    );
+    copyPromises.push(...showcaseCopyPromises);
+    await Promise.all(copyPromises);
+
+    // Delete any existing verified rows
+    if (verifiedAppMetadata) {
+      const deleteVerifiedAppMetadata = await deleteVerifiedAppMetadataSDK(
```

Comment:
why not do this in a single transaction with the update below?

Reply:
Combined the two but note delete_app_metadata_by_pk requires a String! type so we will pass empty string when there's no verifiedAppMetadata and nothing will happen since there are no apps with ID empty string